From b5ca595879cdce2c400b79b1a0b543734295383c Mon Sep 17 00:00:00 2001
From: Jonatan Lindblad <jlindblad@mythtv.org>
Date: Sun, 8 Jun 2014 01:10:42 +0200
Subject: [PATCH 09/23] Internet Content: Fix manual update of RSS
 subscriptions

When triggered from MythFrontend, an RSS update would never complete, leaving
a busy dialog open forever.
---
 mythtv/libs/libmyth/mythrssmanager.cpp |   12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/mythtv/libs/libmyth/mythrssmanager.cpp b/mythtv/libs/libmyth/mythrssmanager.cpp
index df63252..05a1248 100644
--- a/mythtv/libs/libmyth/mythrssmanager.cpp
+++ b/mythtv/libs/libmyth/mythrssmanager.cpp
@@ -92,9 +92,9 @@ void RSSManager::processAndInsertRSS(RSSSite *site)
     {
         // Insert in the DB here.
         insertRSSArticleInDB(site->GetTitle(), *it, site->GetType());
-        m_inprogress.removeOne(site);
     }
 
+    m_inprogress.removeOne(site);
     if (m_inprogress.isEmpty())
         emit finished();
 }
@@ -207,7 +207,10 @@ void RSSSite::process(void)
     m_articleList.clear();
 
     if (!m_data.size())
+    {
+        emit finished(this);
         return;
+    }
 
     QDomDocument domDoc;
 
@@ -215,6 +218,7 @@ void RSSSite::process(void)
     {
         LOG(VB_GENERAL, LOG_ERR, LOC +
             "Failed to set content from downloaded XML");
+        emit finished(this);
         return;
     }
 
@@ -246,11 +250,9 @@ void RSSSite::process(void)
                (*i)->GetSeason(),
                (*i)->GetEpisode(), false));
         }
-        emit finished(this);
     }
     else
-    {
         LOG(VB_GENERAL, LOG_ERR, LOC + "Data is not valid RSS-feed");
-        emit finished(this);
-    }
+
+    emit finished(this);
 }
-- 
1.7.10.2

